tests/mocha/utils_test.js

```diff
@@ -23,73 +23,124 @@ suite('Utils', function() {
-      chai.assert.deepEqual(tokens, ['Hello'], 'No interpolation');
-
-      tokens = Blockly.utils.tokenizeInterpolation('Hello%World');
-      chai.assert.deepEqual(tokens, ['Hello%World'], 'Unescaped %.');
+    setup(function() {
+      this.assertInterpolation = function(string, tokens) {
+        chai.assert.deepEqual(
+            Blockly.utils.tokenizeInterpolation(string), tokens);
+      };
+    });
 
-      tokens = Blockly.utils.tokenizeInterpolation('Hello%%World');
-      chai.assert.deepEqual(tokens, ['Hello%World'], 'Escaped %.');
+    suite('Basic', function() {
+      test('Empty string', function() {
+        this.assertInterpolation('', []);
+      });
 
-      tokens = Blockly.utils.tokenizeInterpolation('Hello %1 World');
-      chai.assert.deepEqual(tokens, ['Hello ', 1, ' World'], 'Interpolation.');
+      test('No interpolation', function() {
+        this.assertInterpolation('Hello', ['Hello']);
+      });
 
-      tokens = Blockly.utils.tokenizeInterpolation('%123Hello%456World%789');
-      chai.assert.deepEqual(tokens, [123, 'Hello', 456, 'World', 789], 'Interpolations.');
+      test('Unescaped %', function() {
+        this.assertInterpolation('Hello%World', ['Hello%World']);
+      });
 
-      tokens = Blockly.utils.tokenizeInterpolation('%%%x%%0%00%01%');
-      chai.assert.deepEqual(tokens, ['%%x%0', 0, 1, '%'], 'Torture interpolations.');
+      test('Escaped percent', function() {
+        this.assertInterpolation('Hello%%World', ['Hello%World']);
+      });
     });
 
-    test('String table', function() {
-      Blockly.Msg = Blockly.Msg || {};
-      Blockly.Msg.STRING_REF = 'test string';
-      var tokens = Blockly.utils.tokenizeInterpolation('%{bky_string_ref}');
-      chai.assert.deepEqual(tokens, ['test string'], 'String table reference, lowercase');
-      tokens = Blockly.utils.tokenizeInterpolation('%{BKY_STRING_REF}');
-      chai.assert.deepEqual(tokens, ['test string'], 'String table reference, uppercase');
-
-      Blockly.Msg.WITH_PARAM = 'before %1 after';
-      tokens = Blockly.utils.tokenizeInterpolation('%{bky_with_param}');
-      chai.assert.deepEqual(tokens, ['before ', 1, ' after'], 'String table reference, with parameter');
-
-      Blockly.Msg.RECURSE = 'before %{bky_string_ref} after';
-      tokens = Blockly.utils.tokenizeInterpolation('%{bky_recurse}');
-      chai.assert.deepEqual(tokens, ['before test string after'], 'String table reference, with subreference');
+    suite('Number interpolation', function() {
+      test('Single-digit number interpolation', function() {
+        this.assertInterpolation('Hello%1World', ['Hello', 1, 'World']);
+      });
+
+      test('Multi-digit number interpolation', function() {
+        this.assertInterpolation(
+            '%123Hello%456World%789', [123, 'Hello', 456, 'World', 789]);
+      });
+
+      test('Crazy interpolation', function() {
+        // No idea what this is supposed to tell you if it breaks. But might
```

Comment:
Percent signs in interpolations, I think? I could see a legitimate need for a % in a message

Reply:
Ohhh you're right. There is a test for an escaped %, but not one where the escaped percent preceeds a number! I'll add a separate test for that so that it's easier to debug hehe